solr/core/src/test/org/apache/solr/search/TestCpuQueryTimeLimit.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.search;
+
+import java.lang.invoke.MethodHandles;
+import java.util.concurrent.TimeUnit;
+import org.apache.solr.client.solrj.SolrClient;
+import org.apache.solr.client.solrj.request.CollectionAdminRequest;
+import org.apache.solr.client.solrj.response.QueryResponse;
+import org.apache.solr.cloud.CloudUtil;
+import org.apache.solr.cloud.MiniSolrCloudCluster;
+import org.apache.solr.cloud.SolrCloudTestCase;
+import org.apache.solr.util.ThreadCpuTime;
+import org.junit.Assume;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TestCpuQueryTimeLimit extends SolrCloudTestCase {
```

Comment:
Is this really SolrCloud related?  Notice how TestCpuTimeSearch does some distributed search yet doesn't need SolrCloud to do it.  I'm trying to keep our tests leaner.

Reply:
I'm not sure if this pseudo-cluster covers the same code paths when making distributed calls.

Reply:
I wouldn't choose SolrCloudTestCase merely because you are not sure of this.  Remember that distributed-search existed before SolrCloud!

Reply:
Thinking of the new SolrClientTestRule stuff; it'd be cool if basic sharded search/indexing could be done to allow a test like the one here to switch between Standalone & SolrCloud trivially.  I'm hoping to get there but the work isn't done yet.  FYI: https://issues.apache.org/jira/browse/SOLR-11872

Reply:
I changed the latest version of the test to use SolrJettyTestRule.

Reply:
Changed it back to SolrCloudTestCase, it was too unstable for distrib multi-shard query.